engine/collection/ingest/rate_limiter.go

```diff
@@ -0,0 +1,138 @@
+package ingest
+
+import (
+	"strings"
+	"sync"
+
+	"golang.org/x/time/rate"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// AddressRateLimiter limits the rate of ingested transactions with a given payer address.
+type AddressRateLimiter struct {
+	mu       sync.RWMutex
+	limiters map[flow.Address]*rate.Limiter
+	limit    rate.Limit // X messages allowed per second
+	burst    int        // X messages allowed at one time
+}
+
+// AddressRateLimiter limits the rate of ingested transactions with a given payer address.
+// It allows the given "limit" amount messages per second with a "burst" amount of messages to be sent at once
+//
+// for example,
+// To config 1 message per 100 milliseconds, convert to per second first, which is 10 message per second,
+// so limit is 10 ( rate.Limit(10) ), and burst is 1.
+// Note: rate.Limit(0.1), burst = 1 means 1 message per 10 seconds, instead of 1 message per 100 milliseconds.
+//
+// To config 3 message per minute, the per-second-basis is 0.05 (3/60), so the limit should be rate.Limit(0.05),
+// and burst is 3.
```

Comment:
```suggestion
// and burst is 3.
//
// Note: The rate limit configured for each node may differ from the effective network-wide rate limit
// for a given payer. In particular, the number of clusters and the message propagation factor will
// influence how the individual rate limit translates to a network-wide rate limit. 
// For example, suppose we have 5 collection clusters and configure each Collection Node with a rate
// limit of 1 message per second. Then, the effective network-wide rate limit for a payer address would
// be *at least* 5 messages per second.
```

Suggesting some documentation based on previous comment: https://github.com/onflow/flow-go/pull/5218/files#r1456473708